pkgs/by-name/ko/koodo-reader/package.nix

```diff
@@ -0,0 +1,120 @@
+{ lib
+, mkYarnPackage
+, fetchFromGitHub
+, applyPatches
+, fetchYarnDeps
+, makeDesktopItem
+, copyDesktopItems
+, wrapGAppsHook
+, gtk3
+, electron
+}:
+
+mkYarnPackage rec {
+  pname = "koodo-reader";
+  version = "1.5.9";
+
+  src = applyPatches {
+    src = fetchFromGitHub {
+      owner = "troyeguo";
+      repo = pname;
+      rev = "v${version}";
+      hash = "sha256-SScbK3pJRqMQ14vj2z4Ek5eSfGxbWN5TYIWLFKRofFk=";
+    };
+    patches = [ ./update-react-i18next.patch ]; # Could be upstreamed
+  };
+
+  packageJSON = ./package.json;
+
+  patches = [ ./fix-isdev.patch ];
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-YDnFJXeebvfpk5TSHS/0mg2INTHt1ZLfoo8lfG/bNgs=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems wrapGAppsHook ];
+
+  dontWrapGApps = true;
+
+  env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  configurePhase = ''
+    cp -r $node_modules node_modules
+    chmod +w node_modules
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    export HOME=$(mktemp -d)
+    yarn --offline build
+    yarn --offline run electron-builder --dir \
+      -c.electronDist=${electron}/libexec/electron \
+      -c.electronVersion=${electron.version}
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 assets/icons/256x256.png $out/share/icons/hicolor/256x256/apps/${pname}.png
+    install -Dm644 ${./mime-types.xml} $out/share/mime/packages/${pname}.xml
```

Comment:
is this file fetched from somewhere? please document if so or consider to `fetchurl` it

Reply:
It would be generated at build time, however, we are building using the `--dir` flag, which skips the platform specific file generation, thus I included it manually.